### PR TITLE
homebrew: fix incorrect handling of aliases

### LINKED ISSUE #9240: homebrew: `state: present` not respected for some packages
### Summary

Since upgrading `community.general` to `10.1.0`, `homebrew` upgrades some packages even when `state` is `present`. I'm aware of two such packages, `sqlite` and `openssl`. Both have aliases, so I suspect the root cause is related to that, and maybe https://github.com/ansible-collections/community.general/pull/9181?

The issue disappears after downgrading `community.general` to 10.0.1

### Issue Type

Bug Report

### Component Name

homebrew

### Ansible Version

```console (paste below)
$ ansible --version

```

### Community.general Version

10.1.0

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

macOS

### Steps to Reproduce

```yaml (paste below)
- name: Install packages
  community.general.homebrew:
    name: ["openssl", readline", "sqlite3", "xz", "zlib", "tcl-tk"]
    state: present
    update_homebrew: false
```

### Expected Results

No change in packages if they are installed

### Actual Results

⚠️ Notice that the package names passed as inputs end up in `changed_pkgs` (and they are indeed upgraded), while their aliases end up in `unchanged_pkgs`

```console (paste below)
2024-12-12T09:17:19.851Z	INFO	TASK [brew-packages : Print install result] ************************************
2024-12-12T09:17:19.851Z	INFO	Thursday 12 December 2024  09:17:19 +0000 (0:00:02.985)       0:10:40.935 ***** 
2024-12-12T09:17:19.868Z	INFO	ok: [192.168.66.2] => {
2024-12-12T09:17:19.868Z	INFO	    "brew_packages_present_result": {
2024-12-12T09:17:19.868Z	INFO	        "attempts": 1,
2024-12-12T09:17:19.868Z	INFO	        "changed": true,
2024-12-12T09:17:19.868Z	INFO	        "changed_pkgs": [
2024-12-12T09:17:19.868Z	INFO	            "openssl",
2024-12-12T09:17:19.868Z	INFO	            "sqlite3"
2024-12-12T09:17:19.868Z	INFO	        ],
2024-12-12T09:17:19.868Z	INFO	        "failed": false,
2024-12-12T09:17:19.868Z	INFO	        "msg": "Changed: 2, Unchanged: 6",
2024-12-12T09:17:19.868Z	INFO	        "unchanged_pkgs": [
2024-12-12T09:17:19.868Z	INFO	            "xz",
2024-12-12T09:17:19.868Z	INFO	            "readline",
2024-12-12T09:17:19.868Z	INFO	            "tcl-tk",
2024-12-12T09:17:19.868Z	INFO	            "zlib",
2024-12-12T09:17:19.868Z	INFO	            "sqlite",
2024-12-12T09:17:19.868Z	INFO	            "openssl@3"
2024-12-12T09:17:19.868Z	INFO	        ]
2024-12-12T09:17:19.868Z	INFO	    }
2024-12-12T09:17:19.868Z	INFO	}
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by UnknownPlatypus**
##### SUMMARY
Fixes #9240 

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
homebrew

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: tests/integration/targets/homebrew/tasks/formulae.yml [hunk 1]
```diff
@@ -321,3 +321,45 @@
...
+      name: "sqlite3"
+      state: present
+      update_homebrew: false
+    become: true
+    become_user: "{{ brew_stat.stat.pw_name }}"
+    register: package_result
+
+  - assert:
+      that:
+        - package_result.changed
```
Comment (russoz) [inline tests/integration/targets/homebrew/tasks/formulae.yml:346]:
```suggestion
        - package_result is changed
```

File: tests/integration/targets/homebrew/tasks/formulae.yml [hunk 2]
```diff
@@ -321,3 +321,45 @@
...
+      name: "sqlite3"
+      state: present
+      update_homebrew: false
+    become: true
+    become_user: "{{ brew_stat.stat.pw_name }}"
+    register: package_result
+
+  - assert:
+      that:
+        - not package_result.changed
```
Comment (russoz) [inline tests/integration/targets/homebrew/tasks/formulae.yml:362]:
```suggestion
        - package_result is not changed
```

File: tests/integration/targets/homebrew/tasks/formulae.yml [hunk 3]
```diff
@@ -321,3 +321,45 @@
...
+        - "package_result.unchanged_pkgs == []"
+
+  - name: Again install sqlite package using alias (sqlite3)
+    homebrew:
+      name: "sqlite3"
+      state: present
+      update_homebrew: false
+    become: true
+    become_user: "{{ brew_stat.stat.pw_name }}"
+    register: package_result
```
Comment (russoz) [inline tests/integration/targets/homebrew/tasks/formulae.yml:358]: I would recommend using a variable name different from the one used in the previous task (line 342), so it becomes clearer to whoever is reading what exactly is being referred to here.

**Review by russoz — COMMENTED** [review]
hi @UnknownPlatypus 

A couple of minor adjustments and I think we're good.

**Comment by UnknownPlatypus** [comment]
@russoz Thanks for teh review, I've also made a little change to handle casks correctly, the issue was flagged by [the previous test run](https://dev.azure.com/ansible/community.general/_build/results?buildId=131190&view=logs&jobId=f13b7acc-ed76-5bbc-cedc-412bd44dee90&j=0cae1a7c-b31e-5ffa-fa7c-644a462c0af7&t=26086552-dcdf-506e-494f-fa8fe2a2637e)

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@UnknownPlatypus thanks for fixing this!
@russoz thanks for reviewing!

**Pull request merged**